airbyte-integrations/connectors/source-datadog/source_datadog/streams.py

```diff
@@ -286,3 +304,112 @@ def next_page_token(self, response: requests.Response) -> Optional[Mapping[str,
+        self.name = name
+        self.data_source = data_source
+        self.query_string = query_string
+
+    @property
+    def http_method(self) -> str:
+        return "POST"
+
+    def path(self, **kwargs) -> str:
+        return "query/timeseries"
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        self._name = value
+
+    @property
+    def cursor_field(self) -> Union[str, List[str]]:
+        return "sync_date"
+
+    def request_body_json(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> Optional[Mapping]:
+
+        if self.query_end_date:
+            end_date = int(parse(self.query_end_date).timestamp() * 1000)
+        else:
+            end_date = int(datetime.now().timestamp()) * 1000
+
+        if self.query_start_date:
+            start_date = int(parse(self.query_start_date).timestamp() * 1000)
+        elif self._cursor_value:
+            start_date = int(parse(self._cursor_value).timestamp() * 1000)
+        else:
+            start_date = int((datetime.now() - timedelta(hours=24)).timestamp()) * 1000
+
+        if self.data_source in ["metrics", "cloud_cost"]:
+            payload = {
+                "data": {
+                    "type": "timeseries_request",
+                    "attributes": {
+                        "to": end_date,
+                        "from": start_date,
+                        "queries": [
+                            {
+                                "data_source": self.data_source,
+                                "query": self.query_string,
+                                "name": self.name,
+                            }
+                        ],
+                    },
+                }
+            }
+        elif self.data_source in ["logs", "rum"]:
+            payload = {
+                "data": {
+                    "type": "timeseries_request",
+                    "attributes": {
+                        "to": end_date,
+                        "from": start_date,
+                        "queries": [
+                            {
+                                "data_source": self.data_source,
+                                "name": self.name,
+                                "search": {
+                                    "query": self.query_string
+                                },
+                                "compute": {
+                                    "aggregation": "count"
+                                },
+                            }
+                        ],
+                    },
+                }
+            }
+        return payload
+
+    def get_json_schema(self) -> Mapping[str, Any]:
+        local_json_schema = {
+            "$schema": "http://json-schema.org/draft-07/schema#",
+            "type": "object",
+            "properties": {},
```

Comment:
I think it's missing to build the actual schema for the response.

Reply:
Do we actually need a specific schema for every response ? I saw generic schemas being used in datadog [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-datadog/source_datadog/schemas/metrics.json)